### PR TITLE
More Resilience

### PR DESCRIPTION
**Pull request by djmitche**
In particular, this will be resilient to 5xx errors.

Would you rather just make retry() work on everything *except* 4xx's?

### CODE REVIEW

**Comment by djmitche** [comment]
Also can you make a release with this change?

File: src/mozillians.js [hunk 1]
```diff
@@ -5,13 +5,17 @@ import url from 'url'
 import _ from 'lodash'
 import crypto from 'crypto'
 
-/** Transient HTTP errors that causes a retry */
+/** Transient error.codes that cause a retry */
```
Comment (jonasfj) [inline src/mozillians.js:8]: Please don't mix http codes and node.js error code...
My suggestion is that we always retry if it's a 5xx :)

File: src/mozillians.js [hunk 2]
```diff
@@ -65,31 +69,35 @@ let request = (options, timeout) => {
...
           return reject(err);
         }
 
+        // Add payload property to response object
+        try {
+          payload = JSON.parse(payload);
+        } catch (err) {
+          let err = new Error("Failed to parse JSON payload: " + payload);
+          err.detail = payload;
+          err.code = 'INVALID_JSON';
```
Comment (jonasfj) [inline src/mozillians.js:97]: Nice

File: src/mozillians.js [hunk 3]
```diff
@@ -65,31 +69,35 @@ let request = (options, timeout) => {
...
+          return eject(err);
```
Comment (jonasfj) [inline src/mozillians.js:98]: `reject`

**Review by jonasfj — CHANGES_REQUESTED** [review]

**Comment by jonasfj** [comment]
I'll fix and land this..

**Comment by jonasfj** [comment]
Done this should be part of `0.1.3`

**Pull request merged**